go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -160,251 +225,261 @@ func getPodMAC(client *corev1client.CoreV1Client, pod *kapi.Pod) (podMAC string,
 		if err != nil {
-			fmt.Printf("execInPod() failed with %s stderr %s stdout %s \n", err, hostError, hostOutput)
-			klog.V(5).Infof("execInPod() failed err %s - podInfo %v - ovnkubePod Name %s", err, podInfo, ovnkubePod.Name)
 			return nil, err
 		}
-
-		klog.V(5).Infof("==>ovnkubePod %s: ip addr show: %q", ovnkubePod.Name, hostOutput)
-
-		var data []IpAddrReq
-		hostOutput = strings.Replace(hostOutput, "\n", "", -1)
-		klog.V(5).Infof("==> host %s NOW: %s", ovnkubePod.Name, hostOutput)
-		err = json.Unmarshal([]byte(hostOutput), &data)
-		if err != nil {
-			klog.V(1).Infof("JSON ERR: couldn't get stuff from data %v; json parse error: %v", data, err)
-			return nil, err
+		name, ok := ovsInterfaceInformation["name"]
+		if !ok {
+			return nil, fmt.Errorf("interface name not found in interface information: %v", ovsInterfaceInformation)
 		}
-		klog.V(5).Infof("Size of IpAddrReq array: %v", len(data))
-		klog.V(5).Infof("IpAddrReq: %v", data)
-
-		for _, addr := range data {
-			if addr.IfIndex == linkIndex {
-				hostInterface = addr.IfName
-				klog.V(5).Infof("ifName: %v\n", addr.IfName)
-				break
-			}
+		ofport, ok := ovsInterfaceInformation["ofport"]
+		if !ok {
+			return nil, fmt.Errorf("interface ofport not found in interface information: %v", ovsInterfaceInformation)
 		}
-		klog.V(5).Infof("hostInterface name is %s\n", hostInterface)
-		podInfo.VethName = hostInterface
+		podInfo.PrimaryInterfaceName = "eth0"
+		podInfo.VethName = strings.Trim(name, "\"")
+		podInfo.OfportNum = ofport
 	}
 
-	// ovs-vsctl get Interface [vethname] ofport
-	portCmd := "ovs-vsctl get Interface " + podInfo.VethName + " ofport"
-	klog.V(5).Infof("Command is: %s", portCmd)
-	portOutput, portError, err := execInPod(coreclient, restconfig, ovnNamespace, ovnkubePod.Name, "ovnkube-node", portCmd, "")
+	podInfo.NodeExternalBridgeName, err = getNodeExternalBridgeName(coreclient, restconfig, ovnNamespace, podInfo.OvnKubePodName, cmd, podInfo.NodeName)
 	if err != nil {
-		fmt.Printf("execInPod() failed with %s stderr %s stdout %s \n", err, portError, portOutput)
-		klog.V(5).Infof("execInPod() failed err %s - podInfo %v - ovnkubePod Name %s", err, podInfo, ovnkubePod.Name)
 		return nil, err
 	}
-	podInfo.PortNum = strings.Replace(portOutput, "\n", "", -1)
 
-	// ovs-vsctl get Interface ovn-k8s-mp0 ofport
-	portCmd = "ovs-vsctl get Interface " + "ovn-k8s-mp0" + " ofport"
-	klog.V(5).Infof("Command is: %s", portCmd)
-	localOutput, localError, err := execInPod(coreclient, restconfig, ovnNamespace, ovnkubePod.Name, "ovnkube-node", portCmd, "")
+	return podInfo, err
+}
+
+// getNodeExternalBridgeName gets the name of the external bridge of this node, e.g. breth0 or br-ex.
+func getNodeExternalBridgeName(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, ovnNamespace, podName, nbcmd, nodeName string) (string, error) {
+	cmd := "ovn-nbctl " + nbcmd + " --bare --no-heading --column=name find Logical_Switch_Port options:network_name=physnet"
```

Comment:
this logic will fail for secondary bridges where we have another external bridge...not many customers use this feature since its not GA but still calling this out.

Reply:
probably the safest way is to get the node's MAC, then do ovs-vsctl --columns=name find bridge other_config:hwaddr='02:42:ac:12:00:04' or something similar